matching-core/src/main/java/com/graphhopper/matching/MapMatching.java

```diff
@@ -124,152 +124,195 @@ public void setMaxVisitedNodes(int maxNodesToVisit) {
     }
 
     /**
-     * This method does the actual map matchting.
+     * This method does the actual map matching.
      * <p>
      * @param gpxList the input list with GPX points which should match to edges
      * of the graph specified in the constructor
      */
     public MatchResult doWork(List<GPXEntry> gpxList) {
         if (gpxList.size() < 2) {
-            throw new IllegalArgumentException("To few coordinates in input file (" + gpxList.size() + "). Correct format?");
+            throw new IllegalArgumentException("Too few coordinates in input file (" + gpxList.size() + "). Correct format?");
         }
 
-        EdgeFilter edgeFilter = new DefaultEdgeFilter(encoder);
-        List<TimeStep<GPXExtension, GPXEntry>> timeSteps = new ArrayList<TimeStep<GPXExtension, GPXEntry>>();
-        List<QueryResult> allCandidates = new ArrayList<QueryResult>();
-        final Map<String, Path> paths = new HashMap<String, Path>();
-        GPXEntry previous = null;
+        final EdgeFilter edgeFilter = new DefaultEdgeFilter(encoder);
+        
+        // Compute all candidates first. Would be great if this could be done on-the-fly instead.
```

Comment:
My goal for this pull request was to create all candidates on the fly, but I couldn't figure out how to do this (see below).